dot/parachain/availability-store/availability_store.go

```diff
@@ -79,13 +80,52 @@ type AvailabilityStoreSubsystem struct {
 	SubSystemToOverseer chan<- any
 	OverseerToSubSystem <-chan any
 	availabilityStore   availabilityStore
-	pruningConfig       pruningConfig
-	clock               subsystemClock
-	//TODO: pruningConfig PruningConfig
-	//TODO: clock         Clock
+	finalizedNumber     parachaintypes.BlockNumber
+	knownBlocks         KnownUnfinalizedBlock
+	pruningConfig       PruningConfig
 	//TODO: metrics       Metrics
 }
 
+// NewAvailabilityStoreSubsystem creates a new instance of AvailabilityStoreSubsystem
+func NewAvailabilityStoreSubsystem(db database.Database) *AvailabilityStoreSubsystem {
+	availabilityStore := NewAvailabilityStore(db)
+
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
I know this doesn't relate to to this PR and I should have mentioned it in the skeleton PR, but I feel pretty strongly that we should shouldn't be adding contexts as attributes of subsystems.  Read this [blog post ](https://dave.cheney.net/2017/08/20/context-isnt-for-cancellation) from Dave Cheney.  I think we should implement the service pattern to explicitly stop subsystems.  I can address this in a PR to the parachain branch.

Reply:
Good point, yes we should implement the service pattern.